src/js/tracks/vtt.js

```diff
@@ -0,0 +1,136 @@
+/**
+ * @file vtt.js
+ */
+import vttjs from 'videojs-vtt.js';
+import Component from '../component.js';
+import mergeOptions from '../utils/merge-options.js';
+import window from 'global/window';
+import document from 'global/document';
+import {isPlainEmpty} from '../utils/obj.js';
+
+let loadedCallback;
+
+export let vttjsLoaded = !isPlainEmpty(vttjs) ||
+                         typeof window.WebVTT === 'function';
+
+/**
+ * The `VttLoader` is the `Component` that decides how to load in vttjs
+ *
+ * @extends Component
+ * @fires VttLoader#vttjsloaded
+ * @fires VttLoader#vttjserror
+ */
+class VttLoader extends Component {
+  constructor(player, options, ready) {
```

Comment:
Having it be a component gives us a lot of things for free. For example, the ability to turn off the inclusion of vttjs when using the novtt build.
There's also precedent with the MediaLoader.